crates/proof-of-sql/src/proof_primitive/dory/dynamic_dory_commitment_helper_gpu.rs

```diff
@@ -7,7 +7,28 @@ use crate::base::{commitment::CommittableColumn, if_rayon, slice_ops::slice_cast
 use blitzar::compute::ElementP2;
 #[cfg(feature = "rayon")]
 use rayon::iter::{IntoParallelIterator, ParallelIterator};
-use tracing::{span, Level};
+use sysinfo::{System, SystemExt};
+use tracing::{span, trace, Level};
+
+#[allow(clippy::cast_precision_loss)]
+fn log_memory_usage(name: &str) {
+    if tracing::level_enabled!(Level::TRACE) {
+        let mut system = System::new_all();
+        system.refresh_memory();
+
+        let available_memory = system.available_memory() as f64 / 1024.0;
+        let used_memory = system.used_memory() as f64 / 1024.0;
+        let percentage_memory_used = (used_memory / (used_memory + available_memory)) * 100.0;
+
+        trace!(
+            "{} Available memory: {:.2} MB, Used memory: {:.2} MB, Percentage memory used: {:.2}%",
+            name,
+            available_memory,
+            used_memory,
+            percentage_memory_used
+        );
```

Comment:
I'm a bit confused. The screenshot looks correct, but the code looks wrong. The `sysinfo` documentation states:
> Returns the amount of available RAM in bytes.

Dividing by 1024 gives KiB, not MiB. So, I'm thinking that the number on the screenshot is KiB, not MB.

But, if that's true, the screenshot implies that there is only 317440.51 KiB = 30 MiB total memory.